staging/src/k8s.io/client-go/tools/leaderelection/leaderelection_test.go

```diff
@@ -917,3 +917,180 @@ func TestTryAcquireOrRenewEndpointsLeases(t *testing.T) {
 func TestTryAcquireOrRenewConfigMapsLeases(t *testing.T) {
 	testTryAcquireOrRenewMultiLock(t, "configmapsleases")
 }
+func TestTryToReleaseEndpointsLeases(t *testing.T) {
+	testTryReleaseMultiLock(t, "endpointsleases")
+}
+func TestTryReleaseConfigMapsLeases(t *testing.T) {
+	testTryReleaseMultiLock(t, "configmapsleases")
+}
+func testTryReleaseMultiLock(t *testing.T, objectType string) {
+	future := time.Now().Add(1000 * time.Hour)
+	primaryType, secondaryType := multiLockType(t, objectType)
+	tests := []struct {
+		name              string
+		observedRecord    rl.LeaderElectionRecord
+		observedRawRecord []byte
+		observedTime      time.Time
+		reactors          []Reactor
+		releaseReactors   []Reactor
+
+		expectSuccess    bool
+		transitionLeader bool
+		doRelease        bool
+		outHolder        string
+	}{
+		{
+			name: "acquire and lease from no object",
+			reactors: []Reactor{
```

Comment:
Generally rather than creating reactors like this we'd have a function that initialized the client, i.e. `newClient: func() *fake.ClientSet { ... create client and setup arbitrary reactors }` which enables you to build more complex setup.